x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts

```diff
@@ -82,12 +83,33 @@ export function initSpacesOnPostAuthRequestInterceptor({
 
         space = await spacesClient.get(spaceId);
       } catch (error) {
+        const wrappedError = wrapError(error);
```

Comment:
> Instead of only responding with the customError when we see a 401 with the WWW-Authenticate header, should we instead only redirect the user to the space selector when we get back a 403 or 404?

I like that idea, I'll make this change